events/equeue/equeue_mbed.cpp

```diff
@@ -154,7 +154,7 @@ bool equeue_sema_wait(equeue_sema_t *s, int ms) {
     if (ms == 0) {
         return false;
     } else if (ms > 0) {
-        timeout.attach_us(callback(equeue_sema_timeout, s), ms*1000);
+        timeout.attach_us(callback(equeue_sema_timeout, s), (us_timestamp_t)ms*1000);
```

Comment:
Shouldn't this be (us_timestamp_t)(ms * 1000) ?

Reply:
True, we can cast ms to us first and then do the multiplication, though I think what i suggested is clearer to the intention but I don't care that much tbh :)

Reply:
I must be missing something but what you suggested is very different and produce different result; see [here](https://godbolt.org/g/NJ1V3Z)